### ISSUE TITLE
Failed to run: streamlit.errors.DuplicateWidgetID: There are multiple widgets with the same `key='init'`

### ISSUE THREAD

**Issue by indrex** [Mohamed-512/Extra-Streamlit-Components#71]
Trying the cookie example from the repo homepage. Streamlit raised the issue:
```
streamlit.errors.DuplicateWidgetID: There are multiple widgets with the same `key='init'`
```

**Comment by elizarp** [comment]
I'm facing the same issue

**Comment by rtyrt** [comment]
Hi @indrex and @elizarp,

Not sure if I'm right or completely sure but I've met the similar issue before and soved it. 
It turns out each streamlit iten is assigned to a key and if you don't pass the key as parameter, it will set as "init" in this package. Also, streamlit seems to be designed with some multithread mechanism that will trigger some thread automatically. If the component is set as a global variable or set at anywhere outside a function, the mechanism may call the init function again (with the same default key), which causes the error message.

So the solution is to put the initialization part into a function and call the function from the beginning of your code to avoid being automatically called by different thread more than once.
(But I still gave up on Extra-Streamlit-Components as my cookie manager )

**Comment by duartemolha** [comment]
@rtyrt what do you use as cookie manager?
I tried this:
https://pypi.org/project/streamlit-cookies-manager/
but it has stopped being developed and I was unable to delete my cookies

This one seems to be working but cannot get rid of the CachedWidgetWarning

**Comment by rtyrt** [comment]
@duartemolha Yeah it's so bad that it has stopped being developed but after trying lots of packages I still think it works the best (Other packages save the "cookie" in the server side and everyone share the same "cookie" which cannot handle oauth login well). And yes It doesn't have the delete function so I compromise on setting the cookie to empty string instead. 

I'm not sure if it's the same issue causing CachedWidgetWarning, but I have modified the following code in the package into this (@st.cache to @st.cache_data) and there's no other error for me:
![image](https://github.com/user-attachments/assets/f12acfb8-65ef-480a-a944-5096c1dfef21)

Hope the information is helpful for you!!
